bpf/kmesh/workload/include/authz.h

```diff
@@ -0,0 +1,213 @@
+                    return MATCHED;
+                }
+            } else {
+                if (bpf_htons(ports[i]) == tuple_info->ipv6.dport) {
+                    return MATCHED;
+                }
+            }
+        }
+    }
+    return UNMATCHED;
+}
+
+static inline int match_check(Istio__Security__Match *match, struct xdp_info *info, struct bpf_sock_tuple *tuple_info)
+{
+    __u32 matchResult;
+
+    // if multiple types are set, they are AND-ed, all matched is a match
+    // todo: add other match types
+    matchResult = matchDstPorts(match, info, tuple_info);
+    return matchResult;
+}
+
+static inline int
+clause_match_check(Istio__Security__Clause *cl, struct xdp_info *info, struct bpf_sock_tuple *tuple_info)
+{
+    void *matchsPtr = NULL;
+    Istio__Security__Match *match = NULL;
+    __u32 i;
+
+    if (cl->n_matches == 0) {
+        return UNMATCHED;
+    }
+    matchsPtr = kmesh_get_ptr_val(cl->matches);
+    if (!matchsPtr) {
+        return MATCHED;
+    }
+
+#pragma unroll
+    for (i = 0; i < MAX_MEMBER_NUM_PER_POLICY; i++) {
+        if (i >= cl->n_matches) {
+            break;
+        }
+        match = (Istio__Security__Match *)kmesh_get_ptr_val((void *)*((__u64 *)matchsPtr + i));
+        if (!match) {
+            continue;
+        }
+        // if any match matches, it is a match
+        if (match_check(match, info, tuple_info) == MATCHED) {
+            return MATCHED;
+        }
+    }
+    return UNMATCHED;
+}
+
+static inline int
+rule_match_check(Istio__Security__Rule *rule, struct xdp_info *info, struct bpf_sock_tuple *tuple_info)
+{
+    void *clausesPtr = NULL;
+    Istio__Security__Clause *clause = NULL;
+    __u32 i;
+
+    if (rule->n_clauses == 0) {
+        BPF_LOG(ERR, AUTH, "rule has no clauses\n");
+        return UNMATCHED;
+    }
+    // Clauses are AND-ed.
+    clausesPtr = kmesh_get_ptr_val(rule->clauses);
+    if (!clausesPtr) {
+        BPF_LOG(ERR, AUTH, "failed to get clauses from rule\n");
+        return UNMATCHED;
+    }
+
+#pragma unroll
+    for (i = 0; i < MAX_MEMBER_NUM_PER_POLICY; i++) {
+        if (i >= rule->n_clauses) {
+            break;
+        }
+        clause = (Istio__Security__Clause *)kmesh_get_ptr_val((void *)*((__u64 *)clausesPtr + i));
+        if (!clause) {
+            continue;
+        }
+        if (clause_match_check(clause, info, tuple_info) == UNMATCHED) {
+            return UNMATCHED;
+        }
+    }
+    return MATCHED;
+}
+
+static inline int
+do_auth(Istio__Security__Authorization *policy, struct xdp_info *info, struct bpf_sock_tuple *tuple_info)
+{
+    void *rulesPtr = NULL;
+    Istio__Security__Rule *rule = NULL;
+    int matchFlag = 0;
+    __u32 i = 0;
+
+    if (policy->n_rules == 0) {
+        BPF_LOG(ERR, AUTH, "auth policy %s has no rules\n", kmesh_get_ptr_val(policy->name));
+        return AUTH_ALLOW;
+    }
+
+    // Rules are OR-ed.
+    rulesPtr = kmesh_get_ptr_val(policy->rules);
```

Comment:
I am not familiar with this, but looks weired to me, It is using the ptr address as key, we cannot assume each time the address of a policy is same.  Especially when a policy is updated